sheerwater_benchmarking/utils/caching.py

```diff
@@ -28,6 +30,129 @@
+                               validate_cache_timeseries=None, retry_null_cache=None):
+    """Reset all global variables to defaults and set the new values."""
+    global global_recompute, global_force_overwrite, \
+        global_validate_cache_timeseries, global_retry_null_cache
+
+    # Simple logic for global variables
+    global_force_overwrite = force_overwrite
+    global_validate_cache_timeseries = validate_cache_timeseries
+    global_retry_null_cache = retry_null_cache
+
+    # More complex logic for recompute
+    if recompute == True:  # noqa: E712
+        global_recompute = False
+    elif isinstance(recompute, str) and recompute != '_all':
+        # if a single function's name is passed, convert to a list
+        global_recompute = [recompute]
+    else:
+        global_recompute = recompute  # if recompute is false, '_all' or a list
+
+
+def check_if_nested_fn():
+    """Check if the current scope downstream from another cached function."""
+    # Get the current frame
+    stack = inspect.stack()
+    # skip the first two frames (this function and the current cacheable function)
+    for frame_info in stack[2:]:
+        frame = frame_info.frame
+        func_name = frame.f_code.co_name
+        if 'wrapper' in func_name:
+            # There is a cachable function upstream of this one
+            return True
+    # No cachable function upstream of this one
+    return False
+
+
+def get_modified_cache_path(cache_path, modification='local'):
+    """Get a modified cache path based on the cache path and the modification type.
+
+    Args:
+        cache_path (str): Path to cache file, either local or GCS path
+        modification (str): The modification to make to the cache path. One of:
+            'local': return the local cache path
+            'temp': return the remote temp cache path
+
+    Returns:
+        str: Local cache path
+    """
+    if cache_path is None:
+        return None
+    if cache_path.startswith(CACHE_ROOT_DIR):
+        cache_key = cache_path.split(CACHE_ROOT_DIR)[1]
+        if modification == 'local':
+            return os.path.join(LOCAL_CACHE_ROOT_DIR, cache_key)
+        elif modification == 'temp':
+            return os.path.join(CACHE_ROOT_DIR, 'temp/', cache_key)
+        else:
+            raise ValueError("Invalid modification type")
+    else:
+        raise ValueError("Cache path must start with CACHE_ROOT_DIR")
+
+
+def sync_local_remote(cache_path=None, verify_path=None, null_path=None):
+    """Sync a remote cache to a local cache.
+
+    Args:
+        cache_path (str): The path to the cache, if None, will not sync the cache
+        verify_path (str): The path to the verify file, if None, will not sync the verify
+        null_path (str): The path to the null cache, if None, will not sync the null
+    """
+    # Check that remote exists
+    fs = fsspec.core.url_to_fs(cache_path, storage_options=CACHE_STORAGE_OPTIONS)[0]
+    if cache_path and not fs.exists(cache_path):
```

Comment:
I think this can be simplified to 
```
if cache_path:
    fs_local.rm(local_cache_path)
    if fs.exists(cache_path):   
        fs.get(cache_path, local_cache_path)````
        
 The question is whether we want to auto-delete these local caches...I think we do?